client/src/App.js

```diff
@@ -1,29 +1,58 @@
-import React from 'react';
-import { MuiThemeProvider } from '@material-ui/core';
-import { BrowserRouter, Route, Switch } from 'react-router-dom';
-import { theme } from './themes/theme';
-import Signup from './pages/signup';
-import Signin from './pages/signin';
+import React, { useContext, useState, useEffect } from "react";
+import { MuiThemeProvider } from "@material-ui/core";
+import { BrowserRouter, Route, Switch } from "react-router-dom";
+import { theme } from "./themes/theme";
+import Signup from "./pages/signup";
+import Signin from "./pages/signin";
 import Dashboard from './pages/Dashboard';
-import Home from './pages/TempHome';
+import FAQ from './pages/FAQ';
+import Blog from './pages/Blog';
 import Profile from './pages/Profile';
+import axios from 'axios';
+import ProtectedRoute from './ProtectedRoute';
+import { UserProvider } from './context/UserContext';
+import { AuthContext } from './context/AuthContext';
 
-import './App.css';
+import "./App.css";
 
 function App() {
-	return (
-		<MuiThemeProvider theme={theme}>
-			<BrowserRouter>
-				<Switch>
-					<Route path="/" component={Home} />
-					<Route path="/profile" component={Profile} />
-					<Route path="/signin" component={Signin} />
-					<Route path="/signup" component={Signup} />
-					<Route path="/dashboard" component={Dashboard} />
-				</Switch>
-			</BrowserRouter>
-		</MuiThemeProvider>
-	);
+  const [authorized, setAuthorized] = useContext(AuthContext);
+
+
+  useEffect(() => {
+    axios.get('/api/JWT')
+      .then(() => {
+        setAuthorized(true)
+      })
+      .catch((err) => {
+        console.log(err)
+        setAuthorized(false)
+      })
+  })
+
+  return (!authorized ?
+    <MuiThemeProvider theme={theme}>
+      <BrowserRouter>
+        <Switch>
+          <Route exact path="/signup" component={Signup} />
+          <Route component={Signin} />
+        </Switch>
+      </BrowserRouter>
+    </MuiThemeProvider>
+    :
+    <MuiThemeProvider theme={theme}>
+      <UserProvider>
+        <BrowserRouter>
+          <Switch>
+            <ProtectedRoute isAuthenticated={authorized} exact path={["/home", "/", "/dashboard"]} component={Dashboard} />
+            <ProtectedRoute isAuthenticated={authorized} path="/faq" component={FAQ} />
+            <ProtectedRoute isAuthenticated={authorized} path="/blog" component={Blog} />
+            <ProtectedRoute isAuthenticated={authorized} path="/profile" component={Profile} />
+          </Switch>
+        </BrowserRouter>
+      </UserProvider>
+    </MuiThemeProvider>
+  );
```

Comment:
This solution mixes methods for handling auth, if you're conditionally rendering the protected routes only if the user is authenticated then they don't need to be protected. I think you can safely remove the ternary logic here since ProtectedRoute handles the redirect already

Reply:
Oh yes, I can see that. The problem relates to what I mentioned above: I can't seem to set authorized in the context to true before the page reloads. Without the conditional rendering, it keeps redirecting to signin. I've moved the API call to ProtectedRoute; I would love any other advice on this.

Reply:
I removed the ProtectedRoute entirely. I've switched it to use only conditional rendering.